cmd/dagger/licenses.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/mitchellh/go-spdx"
+	"github.com/vito/progrock"
+)
+
+const (
+	defaultLicense = "Apache-2.0"
+)
+
+// TODO: dedupe this from Daggerverse, originally hoisted from pkg.go.dev
+var licenseFiles = []string{
+	"COPYING",
+	"COPYING.md",
+	"COPYING.markdown",
+	"COPYING.txt",
+	"LICENCE",
+	"LICENCE.md",
+	"LICENCE.markdown",
+	"LICENCE.txt",
+	"LICENSE",
+	"LICENSE.md",
+	"LICENSE.markdown",
+	"LICENSE.txt",
+	"LICENSE-2.0.txt",
+	"LICENCE-2.0.txt",
+	"LICENSE-APACHE",
+	"LICENCE-APACHE",
+	"LICENSE-APACHE-2.0.txt",
+	"LICENCE-APACHE-2.0.txt",
+	"LICENSE-MIT",
+	"LICENCE-MIT",
+	"LICENSE.MIT",
+	"LICENCE.MIT",
+	"LICENSE.code",
+	"LICENCE.code",
+	"LICENSE.docs",
+	"LICENCE.docs",
+	"LICENSE.rst",
+	"LICENCE.rst",
+	"MIT-LICENSE",
+	"MIT-LICENCE",
+	"MIT-LICENSE.md",
+	"MIT-LICENCE.md",
+	"MIT-LICENSE.markdown",
+	"MIT-LICENCE.markdown",
+	"MIT-LICENSE.txt",
+	"MIT-LICENCE.txt",
+	"MIT_LICENSE",
+	"MIT_LICENCE",
+	"UNLICENSE",
+	"UNLICENCE",
+}
+
+func findOrCreateLicense(ctx context.Context, dir string) error {
+	rec := progrock.FromContext(ctx)
+
+	id := licenseID
+	if id == "" {
+		if foundLicense, err := searchForLicense(moduleRoot); err == nil {
+			rec.Debug("found existing LICENSE file", progrock.Labelf("path", foundLicense))
+			return nil
+		}
+
+		id = defaultLicense
+	}
+
+	rec.Warn("no LICENSE file found; generating one for you, feel free to change or remove",
+		progrock.Labelf("license", id))
+
+	license, err := spdx.License(id)
+	if err != nil {
+		return fmt.Errorf("failed to get license: %w", err)
+	}
+
+	if err := os.WriteFile("LICENSE", []byte(license.Text), 0600); err != nil {
```

Comment:
Is this the right default behavior if you are running e.g. `dagger mod init -m ../foo/bar ...`; should we always make it in the cwd?

Reply:
Whoops, missed a couple spots here post-refactor, fixing up - thanks! This and `findOrCreateLicense` should both be using the `dir` arg which is currently ignored.

Reply:
Fixed + added test